// @uifabric/experiments/lib-amd/components/Button/ButtonVariants/MessageBarButton.js
define(["require", "exports", "@uifabric/foundation/lib/next/composed", "../Button.state", "../Button.styles", "../Button.view"], function (require, exports, composed_1, Button_state_1, Button_styles_1, Button_view_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /* eslint-disable deprecation/deprecation */
    var baseTokens = function (props, theme) {
        var palette = theme.palette;
        return {
            borderColor: palette.neutralTertiaryAlt,
            borderColorHovered: palette.neutralTertiaryAlt,
            borderColorPressed: palette.neutralTertiaryAlt,
            contentPadding: 0,
            cursor: 'pointer',
            height: 24,
            minHeight: 24,
            minWidth: 84,
            width: 84,
        };
    };
    var disabledTokens = function (props, theme) {
        var semanticColors = theme.semanticColors;
        return {
            borderColor: semanticColors.buttonBorderDisabled,
            borderColorHovered: semanticColors.buttonBorderDisabled,
            borderColorPressed: semanticColors.buttonBorderDisabled,
            cursor: 'default',
        };
    };
    /** @deprecated */
    exports.MessageBarButtonTokens = function (props, theme) {
        var regularTokens = Button_styles_1.ButtonTokens(props, theme);
        return [regularTokens, baseTokens, props.disabled && disabledTokens];
    };
    /**
     * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
     * future.
     */
    exports.MessageBarButton = composed_1.composed({
        displayName: 'MessageBarButton',
        slots: Button_view_1.ButtonSlots,
        state: Button_state_1.useButtonState,
        styles: Button_styles_1.ButtonStyles,
        tokens: exports.MessageBarButtonTokens,
        view: Button_view_1.ButtonView,
    });
});
