// @uifabric/experiments/lib-amd/components/Button/ButtonVariants/PrimaryButton.js
define(["require", "exports", "@uifabric/foundation/lib/next/composed", "../Button.state", "../Button.styles", "../Button.view"], function (require, exports, composed_1, Button_state_1, Button_styles_1, Button_view_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /* eslint-disable deprecation/deprecation */
    /** @deprecated */
    exports.PrimaryButtonTokens = function (props, theme) { return [
        Button_styles_1.baseTokens,
        !!props.href && Button_styles_1.hrefTokens,
        Button_styles_1.primaryEnabledTokens,
        props.circular && Button_styles_1.circularTokens,
        props.circular && Button_styles_1.primaryCircularTokens,
        props.checked && Button_styles_1.primaryCheckedTokens,
        props.disabled && Button_styles_1.disabledTokens,
    ]; };
    /**
     * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
     * future.
     */
    exports.PrimaryButton = composed_1.composed({
        displayName: 'PrimaryButton',
        slots: Button_view_1.ButtonSlots,
        state: Button_state_1.useButtonState,
        styles: Button_styles_1.ButtonStyles,
        tokens: exports.PrimaryButtonTokens,
        view: Button_view_1.ButtonView,
    });
});
