// @uifabric/experiments/lib-commonjs/components/Pagination/Pagination.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Styling_1 = require("../../Styling");
var GlobalClassNames = {
    root: 'ms-Pagination-container',
    pageNumber: 'ms-Pagination-pageNumber',
};
function getStyles(props) {
    var theme = props.theme, format = props.format;
    var palette = theme.palette;
    var classNames = Styling_1.getGlobalClassNames(GlobalClassNames, theme);
    var buttonStyles = {
        cursor: 'pointer',
        border: 'none',
        backgroundColor: 'transparent',
    };
    return {
        root: [
            classNames.root,
            format === 'buttons' && {
                listStyle: 'none',
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
            },
        ],
        previousNextPage: [
            {
                color: palette.themePrimary,
            },
        ],
        previousNextPageDisabled: [
            {
                cursor: 'default',
                backgroundColor: 'transparent',
            },
        ],
        pageNumber: [
            classNames.pageNumber,
            buttonStyles,
            {
                verticalAlign: 'middle',
                minWidth: '32px',
                minHeight: '32px',
                color: palette.black,
                selectors: {
                    '&[aria-selected=true]': {
                        color: palette.blue,
                        cursor: 'default',
                        fontWeight: 'bold',
                        textDecoration: 'underline',
                    },
                    '&:hover[aria-selected=true]': {
                        color: palette.blue,
                        backgroundColor: 'transparent',
                    },
                    ':active': {
                        backgroundColor: 'transparent',
                    },
                },
            },
        ],
        visibleItemLabel: [
            {
                color: palette.neutralSecondary,
            },
        ],
        comboBox: {
            maxWidth: '70px',
            display: 'inline-block',
            verticalAlign: 'middle',
        },
    };
}
exports.getStyles = getStyles;
