// @uifabric/azure-themes/lib-amd/azure/styles/HoverCard.styles.js
define(["require", "exports", "../Constants"], function (require, exports, StyleConstants) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.ExpandingCardStyles = function (props) {
        var theme = props.theme;
        var semanticColors = theme.semanticColors;
        return {
            root: {
                selectors: {
                    '.ms-Callout-main': {
                        borderColor: semanticColors.inputBorder,
                        borderStyle: StyleConstants.borderSolid,
                        borderWidth: StyleConstants.borderWidth,
                    },
                },
            },
        };
    };
    exports.PlainCardStyles = function (props) {
        var theme = props.theme;
        var semanticColors = theme.semanticColors;
        return {
            root: {
                selectors: {
                    '.ms-Callout-main': {
                        borderColor: semanticColors.inputBorder,
                        borderStyle: StyleConstants.borderSolid,
                        borderWidth: StyleConstants.borderWidth,
                    },
                },
            },
        };
    };
});
