// @uifabric/azure-themes/lib-amd/azure/styles/Link.styles.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.LinkStyles = function (props) {
        var isDisabled = props.isDisabled, theme = props.theme;
        var semanticColors = theme.semanticColors;
        var extendedSemanticColors = semanticColors;
        return {
            root: [
                !isDisabled && {
                    color: semanticColors.link,
                    textDecoration: extendedSemanticColors.listUnderline,
                    selectors: {
                        '&:active, &:hover, &:active:hover': {
                            backgroundColor: extendedSemanticColors.linkBackgroundHovered,
                            color: semanticColors.linkHovered,
                        },
                        '&:active': {
                            backgroundColor: extendedSemanticColors.linkBackgroundHovered,
                            border: extendedSemanticColors.choiceGroupContainerBorder + "\n              " + extendedSemanticColors.linkBorderStyle + "\n              " + extendedSemanticColors.linkHovered,
                            color: semanticColors.linkHovered,
                        },
                    },
                },
            ],
        };
    };
});
