// @uifabric/azure-themes/lib-amd/azure/styles/Slider.styles.js
define(["require", "exports", "../Constants"], function (require, exports, StyleConstants) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var SLIDER_BOX_DIMENSION = 8;
    var SLIDER_DIAMETER = 16;
    var SLIDER_OFFSET = 2;
    var SLIDER_BORDER_RADIUS = 0;
    exports.SliderStyles = function (props) {
        var disabled = props.disabled, theme = props.theme, vertical = props.vertical;
        var semanticColors = theme.semanticColors;
        return {
            activeSection: [
                !disabled && {
                    backgroundColor: semanticColors.controlAccent,
                },
                disabled && {
                    background: semanticColors.disabledBodyText,
                },
            ],
            inactiveSection: [
                !disabled && {
                    background: semanticColors.controlOutlineDisabled,
                },
                disabled && {
                    background: semanticColors.disabledBodyText,
                },
            ],
            slideBox: [
                !disabled && {
                    selectors: {
                        '.ms-Slider-thumb': [
                            {
                                borderColor: semanticColors.controlAccent,
                                height: SLIDER_DIAMETER,
                                width: SLIDER_DIAMETER,
                            },
                            vertical && {
                                marginLeft: SLIDER_OFFSET,
                            },
                            !vertical && {
                                marginTop: SLIDER_OFFSET,
                            },
                        ],
                    },
                },
                disabled && {
                    selectors: {
                        '.ms-Slider-thumb': [
                            {
                                borderColor: semanticColors.disabledBodyText,
                                height: SLIDER_DIAMETER,
                                width: SLIDER_DIAMETER,
                            },
                            vertical && {
                                marginLeft: SLIDER_OFFSET,
                            },
                            !vertical && {
                                marginTop: SLIDER_OFFSET,
                            },
                        ],
                    },
                },
            ],
            line: [
                !vertical && {
                    selectors: {
                        '.ms-Slider-active': {
                            height: SLIDER_BOX_DIMENSION,
                            borderRadius: SLIDER_BORDER_RADIUS,
                        },
                        '.ms-Slider-inactive': {
                            height: SLIDER_BOX_DIMENSION,
                            borderRadius: SLIDER_BORDER_RADIUS,
                        },
                    },
                },
                vertical && {
                    selectors: {
                        '.ms-Slider-active': {
                            width: SLIDER_BOX_DIMENSION,
                            borderRadius: SLIDER_BORDER_RADIUS,
                        },
                        '.ms-Slider-inactive': {
                            width: SLIDER_BOX_DIMENSION,
                            borderRadius: SLIDER_BORDER_RADIUS,
                        },
                    },
                },
            ],
            valueLabel: [
                !disabled && {
                    borderColor: semanticColors.inputBorder,
                    borderStyle: StyleConstants.borderSolid,
                    borderWidth: StyleConstants.borderWidth,
                    textAlign: StyleConstants.textAlignCenter,
                    selectors: {
                        '&.ms-Slider-value': {
                            color: semanticColors.bodyText,
                        },
                    },
                },
                disabled && {
                    borderColor: semanticColors.inputBorder,
                    borderStyle: StyleConstants.borderSolid,
                    borderWidth: StyleConstants.borderWidth,
                    color: semanticColors.disabledText,
                    textAlign: StyleConstants.textAlignCenter,
                    selectors: {
                        '&.ms-Slider-value': {
                            color: semanticColors.disabledBodyText,
                        },
                    },
                },
            ],
        };
    };
});
