// @uifabric/experiments/lib-amd/components/MicroFeedback/MicroFeedback.styles.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var baseTokens = {
        questionMargin: '20px 12px 6px',
    };
    var inlineTokens = function (props, theme) {
        return {
            followUpBackgroundColor: theme.palette.white,
            questionMargin: '12px 12px 6px',
            width: '100%',
        };
    };
    exports.MicroFeedbackTokens = function (props, theme) { return [
        baseTokens,
        props.inline && inlineTokens,
    ]; };
    exports.MicroFeedbackStyles = function (props, theme, tokens) {
        return {
            root: {
                margin: '4px 8px 4px 8px',
                width: tokens.width,
            },
            iconContainer: {
                float: 'right',
            },
            followUpQuestion: {
                fontFamily: "'Segoe UI Semibold'",
                margin: tokens.questionMargin,
                fontSize: '12px',
                letterSpacing: '0.1px',
                border: '0px',
            },
            followUpContainer: {
                backgroundColor: tokens.followUpBackgroundColor,
                paddingBottom: '6px',
            },
            thanksContainer: {
                backgroundColor: theme.palette.white,
                padding: '6px',
            },
            followUpOption: {
                paddingLeft: '12px',
                paddingRight: '12px',
                background: 'transparent',
                border: '0px',
                minHeight: 0,
                width: '100%',
            },
            followUpOptionText: {
                fontFamily: "'Segoe UI'",
                fontSize: '12px',
                letterSpacing: '0.11px',
                textAlign: 'left',
                border: '0px',
                width: '100%',
            },
        };
    };
});
