// @uifabric/azure-themes/lib-commonjs/azure/styles/Callout.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AzureDepths_1 = require("../AzureDepths");
exports.CalloutContentStyles = function (props) {
    var theme = props.theme;
    var semanticColors = theme.semanticColors;
    var extendedSemanticColors = semanticColors;
    return {
        root: {
            boxShadow: AzureDepths_1.Depths.depth8,
        },
        calloutMain: {
            color: semanticColors.bodyText,
            fontSize: theme.fonts.medium.fontSize,
            backgroundColor: extendedSemanticColors.controlBackground,
            border: extendedSemanticColors.choiceGroupContainerBorder + "\n      " + extendedSemanticColors.callOutBorderStyle + "\n      " + extendedSemanticColors.primaryButtonBorder,
        },
        beak: {
            backgroundColor: extendedSemanticColors.controlBackground,
            border: extendedSemanticColors.choiceGroupContainerBorder + "\n      " + extendedSemanticColors.callOutBorderStyle + "\n      " + extendedSemanticColors.primaryButtonBorder,
        },
    };
};
