// @uifabric/azure-themes/lib-commonjs/azure/styles/ColorPicker.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AzureDepths_1 = require("../AzureDepths");
exports.ColorPickerStyles = function (props) {
    return {
        input: {
            selectors: {
                '&.ms-TextField': {
                    paddingRight: 4,
                },
                '.ms-TextField-field': {
                    minWidth: 'auto',
                    padding: 5,
                    textOverflow: 'clip',
                },
            },
        },
        table: {
            selectors: {
                'tbody td:last-of-type .ms-ColorPicker-input': {
                    paddingRight: 0,
                },
            },
        },
        tableHeader: {
            selectors: {
                td: {
                    paddingBottom: 4,
                },
            },
        },
    };
};
exports.ColorRectangleStyles = function (props) {
    var theme = props.theme;
    var semanticColors = theme.semanticColors;
    return {
        root: {
            border: "1px solid " + semanticColors.controlOutline,
        },
        thumb: {
            borderColor: semanticColors.controlOutline,
            boxShadow: AzureDepths_1.Depths.depth8,
        },
    };
};
exports.ColorSliderStyles = function (props) {
    var theme = props.theme;
    var semanticColors = theme.semanticColors;
    return {
        sliderThumb: {
            borderColor: semanticColors.controlOutline,
            boxShadow: AzureDepths_1.Depths.depth8,
        },
    };
};
