// @uifabric/azure-themes/lib-commonjs/azure/styles/HoverCard.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var StyleConstants = require("../Constants");
exports.ExpandingCardStyles = function (props) {
    var theme = props.theme;
    var semanticColors = theme.semanticColors;
    return {
        root: {
            selectors: {
                '.ms-Callout-main': {
                    borderColor: semanticColors.inputBorder,
                    borderStyle: StyleConstants.borderSolid,
                    borderWidth: StyleConstants.borderWidth,
                },
            },
        },
    };
};
exports.PlainCardStyles = function (props) {
    var theme = props.theme;
    var semanticColors = theme.semanticColors;
    return {
        root: {
            selectors: {
                '.ms-Callout-main': {
                    borderColor: semanticColors.inputBorder,
                    borderStyle: StyleConstants.borderSolid,
                    borderWidth: StyleConstants.borderWidth,
                },
            },
        },
    };
};
