// @uifabric/experiments/lib/components/FolderCover/initializeFolderCovers.js
import * as React from 'react';
import { registerIcons } from '@uifabric/styling';
var ASSET_CDN_BASE_URL = 'https://spoprod-a.akamaihd.net/files/fabric-cdn-prod_20200708.002/office-ui-fabric-react-assets/foldericons';
export function initializeFolderCovers(baseUrl, options) {
    if (baseUrl === void 0) { baseUrl = ASSET_CDN_BASE_URL; }
    registerIcons({
        fontFace: {},
        style: {
            width: 118,
            height: 86,
            overflow: 'hidden',
        },
        icons: {
            folderCoverLargeDefaultFront: React.createElement("img", { src: baseUrl + "/lg-fg.svg" }),
            folderCoverLargeDefaultBack: React.createElement("img", { src: baseUrl + "/lg-bg.svg" }),
            folderCoverLargeLinkedFront: React.createElement("img", { src: baseUrl + "/lg-fg-linked.svg" }),
            folderCoverLargeLinkedBack: React.createElement("img", { src: baseUrl + "/lg-bg.svg" }),
            folderCoverLargeMediaFront: React.createElement("img", { src: baseUrl + "/lg-fg-media.svg" }),
            folderCoverLargeMediaBack: React.createElement("img", { src: baseUrl + "/lg-bg.svg" }),
        },
    }, options);
    registerIcons({
        fontFace: {},
        style: {
            width: 78,
            height: 58,
            overflow: 'hidden',
        },
        icons: {
            folderCoverSmallDefaultFront: React.createElement("img", { src: baseUrl + "/sm-fg.svg" }),
            folderCoverSmallDefaultBack: React.createElement("img", { src: baseUrl + "/sm-bg.svg" }),
            folderCoverSmallLinkedFront: React.createElement("img", { src: baseUrl + "/sm-fg-linked.svg" }),
            folderCoverSmallLinkedBack: React.createElement("img", { src: baseUrl + "/sm-bg.svg" }),
            folderCoverSmallMediaFront: React.createElement("img", { src: baseUrl + "/sm-fg-media.svg" }),
            folderCoverSmallMediaBack: React.createElement("img", { src: baseUrl + "/sm-bg.svg" }),
        },
    }, options);
}
