// @uifabric/experiments/lib/components/Button/ButtonVariants/DefaultButton.js
// Temporary import file to experiment with next version of foundation.
import { composed } from '@uifabric/foundation/lib/next/composed';
import { useButtonState as state } from '../Button.state';
import { baseTokens, checkedTokens, circularTokens, disabledTokens, enabledTokens, hrefTokens, ButtonStyles as styles, } from '../Button.styles';
import { ButtonSlots as slots, ButtonView as view } from '../Button.view';
/* eslint-disable deprecation/deprecation */
/** @deprecated */
export var DefaultButtonTokens = function (props, theme) { return [
    baseTokens,
    !!props.href && hrefTokens,
    !props.disabled && enabledTokens,
    props.circular && circularTokens,
    props.checked && checkedTokens,
    props.disabled && disabledTokens,
]; };
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
export var DefaultButton = composed({
    displayName: 'DefaultButton',
    slots: slots,
    state: state,
    styles: styles,
    tokens: DefaultButtonTokens,
    view: view,
});
