// @uifabric/experiments/lib/components/Button/ButtonVariants/PrimaryButton.js
// Temporary import file to experiment with next version of foundation.
import { composed } from '@uifabric/foundation/lib/next/composed';
import { useButtonState as state } from '../Button.state';
import { baseTokens, circularTokens, disabledTokens, hrefTokens, primaryCheckedTokens, primaryCircularTokens, primaryEnabledTokens, ButtonStyles as styles, } from '../Button.styles';
import { ButtonSlots as slots, ButtonView as view } from '../Button.view';
/* eslint-disable deprecation/deprecation */
/** @deprecated */
export var PrimaryButtonTokens = function (props, theme) { return [
    baseTokens,
    !!props.href && hrefTokens,
    primaryEnabledTokens,
    props.circular && circularTokens,
    props.circular && primaryCircularTokens,
    props.checked && primaryCheckedTokens,
    props.disabled && disabledTokens,
]; };
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
export var PrimaryButton = composed({
    displayName: 'PrimaryButton',
    slots: slots,
    state: state,
    styles: styles,
    tokens: PrimaryButtonTokens,
    view: view,
});
