// @uifabric/experiments/lib-amd/components/CollapsibleSection/CollapsibleSectionTitle.styles.js
define(["require", "exports", "office-ui-fabric-react"], function (require, exports, office_ui_fabric_react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.getStyles = function (props, theme) { return ({
        root: [
            office_ui_fabric_react_1.getFocusStyle(theme),
            theme.fonts.medium,
            {
                width: '100%',
                display: 'flex',
                alignItems: 'center',
                background: 'none',
                border: 'none',
                color: 'inherit',
                height: 24,
                margin: 0,
                paddingLeft: 4 + (props.indent || 0) * 18,
                selectors: {
                    ':hover': {
                        background: theme.palette.neutralLight,
                    },
                },
            },
        ],
        chevron: [
            {
                flexShrink: 0,
                padding: 0,
                marginRight: 8,
                transition: 'transform .1s linear',
            },
            props.collapsed && {
                transform: 'rotate(-90deg)',
            },
        ],
        text: theme.fonts.small,
    }); };
});
