// @uifabric/experiments/lib-commonjs/components/FloatingSuggestionsComposite/FloatingSuggestionsHeaderFooterItem/FloatingSuggestionsHeaderFooterItem.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var styling_1 = require("@uifabric/styling");
var GlobalClassNames = {
    actionButton: 'ms-FloatingSuggestionsHeaderFooterItem-actionButton',
    buttonSelected: 'ms-FloatingSuggestionsHeaderFooterItem-buttonSelected',
};
exports.getStyles = function (props) {
    var _a, _b, _c;
    var theme = styling_1.getTheme();
    if (!theme) {
        throw new Error('theme is undefined or null in FloatingSuggestionsItem getStyles function.');
    }
    var palette = theme.palette;
    var classNames = styling_1.getGlobalClassNames(GlobalClassNames, theme);
    return {
        actionButton: [
            classNames.actionButton,
            {
                width: '100%',
                padding: '0px',
                minWidth: '0',
                height: '100%',
                selectors: (_a = {},
                    _a[styling_1.HighContrastSelector] = {
                        color: 'WindowText',
                    },
                    _a['&:hover'] = {
                        background: palette.neutralLighter,
                    },
                    _a),
            },
        ],
        buttonSelected: [
            classNames.buttonSelected,
            {
                background: palette.themeLighter,
                selectors: (_b = {
                        '&:hover': (_c = {
                                background: palette.themeLight
                            },
                            _c[styling_1.HighContrastSelector] = tslib_1.__assign({ background: 'Highlight', color: 'HighlightText' }, styling_1.getHighContrastNoAdjustStyle()),
                            _c)
                    },
                    _b[styling_1.HighContrastSelector] = tslib_1.__assign({ background: 'Highlight', color: 'HighlightText' }, styling_1.getHighContrastNoAdjustStyle()),
                    _b),
            },
        ],
    };
};
