// @uifabric/experiments/lib-commonjs/components/FloatingSuggestionsComposite/FloatingPeopleSuggestions/FloatingPeopleSuggestionItems/SuggestionItemDefault.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var styling_1 = require("@uifabric/styling");
var GlobalClassNames = {
    root: 'ms-FloatingPeopleSuggestions-SuggestionItemDefault-persona',
    callout: 'ms-FloatingPeopleSuggestions-SuggestionItemDefault-personaContent',
};
exports.getStyles = function (props) {
    var _a;
    var theme = styling_1.getTheme();
    if (!theme) {
        throw new Error('theme is undefined or null in Editing item getStyles function.');
    }
    // const { semanticColors } = theme;
    var classNames = styling_1.getGlobalClassNames(GlobalClassNames, theme);
    return {
        persona: [
            classNames.root,
            {
                width: '100%',
                selectors: (_a = {},
                    _a['.ms-Persona-detail'] = {
                        minHeight: '40px',
                    },
                    _a),
            },
        ],
        personaContent: [
            classNames.callout,
            {
                display: 'flex',
                width: '100%',
                justifyContent: 'space-between',
                alignItems: 'center',
                padding: '7px 12px',
            },
        ],
    };
};
