// @uifabric/theme-samples/lib-amd/DarkCustomizations/styles/DatePickerStyles.js
define(["require", "exports", "tslib", "../DarkCustomizations"], function (require, exports, tslib_1, DarkCustomizations_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.DatePickerStyles = function (props) {
        var disabled = props.disabled;
        var TextHoverStyle = function () {
            return {
                color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                backgroundColor: DarkCustomizations_1.DarkTheme.palette.neutralQuaternary,
            };
        };
        var TodayAndSelectedDayStyle = function () {
            return {
                '.ms-DatePicker-day-button.ms-DatePicker-day--today': {
                    backgroundColor: DarkCustomizations_1.DarkTheme.palette.themePrimary,
                    color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                },
                '.ms-DatePicker-day-button.ms-DatePicker-day--today:active': {
                    backgroundColor: DarkCustomizations_1.DarkTheme.palette.themePrimary,
                },
                '.ms-DatePicker-day-button.ms-DatePicker-day--today:hover': {
                    color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                },
                '.ms-DatePicker-day--highlighted': {
                    backgroundColor: DarkCustomizations_1.DarkTheme.palette.themeSecondary,
                    color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                },
                '.ms-DatePicker-day--highlighted > .ms-DatePicker-day-button': {
                    backgroundColor: DarkCustomizations_1.DarkTheme.palette.themePrimary,
                },
                '.ms-DatePicker-day--highlighted > .ms-DatePicker-day--today': {
                    color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                },
            };
        };
        var HoverStyles = function () {
            return {
                '.ms-DatePicker-day--highlighted:hover': {
                    backgroundColor: DarkCustomizations_1.DarkTheme.semanticColors.listItemBackgroundChecked,
                    color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                },
                '.ms-DatePicker-day--highlighted > .ms-DatePicker-day-button:hover': {
                    color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                },
                '.ms-DatePicker-day--infocus:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-currentDecade:hover': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary },
                '.ms-DatePicker-day--outfocus:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-monthAndYear:hover': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary },
                '.ms-DatePicker-weekday:hover': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary },
                '.ms-DatePicker-monthOption:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-header > .ms-DatePicker-currentYear:hover': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary },
                '.ms-DatePicker-prevMonth:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-nextMonth:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-prevYear:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-nextYear:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-goToday:hover': {
                    color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                    backgroundColor: DarkCustomizations_1.DarkTheme.palette.themeSecondary,
                },
                '.ms-DatePicker-yearOption:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-prevDecade:hover': tslib_1.__assign({}, TextHoverStyle()),
                '.ms-DatePicker-nextDecade:hover': tslib_1.__assign({}, TextHoverStyle()),
            };
        };
        return {
            callout: {
                backgroundColor: DarkCustomizations_1.DarkTheme.semanticColors.bodyBackground,
                color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                selectors: tslib_1.__assign(tslib_1.__assign({ '.ms-DatePicker-day--infocus': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-currentDecade': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-day--outfocus': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-monthAndYear': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-weekday': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-monthOption': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-currentYear': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-prevMonth': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-nextMonth': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-prevYear': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-nextYear': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-prevDecade': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-nextDecade': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-goToday': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-goToday[disabled]': { display: 'none' }, '.ms-DatePicker-yearOption': { color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary }, '.ms-DatePicker-yearOption--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-monthOption--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-day--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-nextDecade--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-prevDecade--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-prevYear--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-nextYear--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-prevMonth--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary }, '.ms-DatePicker-nextMonth--disabled': { color: DarkCustomizations_1.DarkTheme.palette.neutralSecondary } }, TodayAndSelectedDayStyle()), HoverStyles()),
            },
            icon: [
                {
                    color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                },
                disabled && {
                    color: DarkCustomizations_1.DarkTheme.semanticColors.disabledBodyText,
                },
            ],
            root: [
                disabled && {
                    color: DarkCustomizations_1.DarkTheme.semanticColors.disabledBodyText,
                },
            ],
        };
    };
});
