// @uifabric/experiments/lib-commonjs/components/Button/SplitButton/SplitButton.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Temporary import file to experiment with next version of foundation.
var composed_1 = require("@uifabric/foundation/lib-commonjs/next/composed");
var SplitButton_state_1 = require("./SplitButton.state");
var SplitButton_styles_1 = require("./SplitButton.styles");
var SplitButton_view_1 = require("./SplitButton.view");
/* eslint-disable deprecation/deprecation */
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
exports.SplitButton = composed_1.composed({
    displayName: 'SplitButton',
    slots: SplitButton_view_1.SplitButtonSlots,
    state: SplitButton_state_1.useSplitButtonState,
    styles: SplitButton_styles_1.SplitButtonStyles,
    tokens: SplitButton_styles_1.SplitButtonTokens,
    view: SplitButton_view_1.SplitButtonView,
});
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
exports.default = exports.SplitButton;
