// @uifabric/experiments/lib-commonjs/components/UnifiedPicker/UnifiedPicker.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var styling_1 = require("@uifabric/styling");
var GlobalClassNames = {
    pickerText: 'ms-UnifiedPicker-text',
    pickerInput: 'ms-UnifiedPicker-input',
    pickerDiv: 'ms-UnifiedPicker-div',
    selectionZone: 'ms-UnifiedPicker-selectionZone',
};
exports.getStyles = function (props) {
    var theme = styling_1.getTheme();
    if (!theme) {
        throw new Error('theme is undefined or null in Editing item getStyles function.');
    }
    // const { semanticColors } = theme;
    var _a = theme.palette, neutralTertiary = _a.neutralTertiary, themeLight = _a.themeLight;
    var classNames = styling_1.getGlobalClassNames(GlobalClassNames, theme);
    return {
        pickerText: [
            classNames.pickerText,
            {
                display: 'flex',
                flex: '1 1 auto',
                flexWrap: 'wrap',
                alignItems: 'center',
                boxSizing: 'border-box',
                border: "1px solid " + neutralTertiary,
                minWidth: '180px',
                padding: '1px',
                minHeight: '32px',
                selectors: {
                    '&:hover': {
                        borderColor: themeLight,
                    },
                },
            },
        ],
        pickerInput: [
            classNames.pickerInput,
            {
                display: 'flex',
                flex: '1 1 auto',
                height: '34px',
                border: 'none',
                flexGrow: '1',
                outline: 'none',
                padding: '0 6px 0px',
                margin: '1px',
                selectors: {
                    '&::-ms-clear': {
                        display: 'none',
                    },
                },
            },
        ],
        pickerDiv: [
            classNames.pickerDiv,
            {
                display: 'flex',
                flex: '1 1 auto',
            },
        ],
        selectionZone: [
            classNames.selectionZone,
            {
                display: 'flex',
                flex: '1 1 auto',
            },
        ],
    };
};
