// @uifabric/experiments/lib-amd/components/Button/MenuButton/MenuButton.js
define(["require", "exports", "@uifabric/foundation/lib/next/composed", "./MenuButton.state", "./MenuButton.styles", "./MenuButton.view"], function (require, exports, composed_1, MenuButton_state_1, MenuButton_styles_1, MenuButton_view_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /* eslint-disable deprecation/deprecation */
    /**
     * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
     * future.
     */
    exports.MenuButton = composed_1.composed({
        displayName: 'MenuButton',
        slots: MenuButton_view_1.MenuButtonSlots,
        state: MenuButton_state_1.useMenuButtonState,
        styles: MenuButton_styles_1.MenuButtonStyles,
        tokens: MenuButton_styles_1.MenuButtonTokens,
        view: MenuButton_view_1.MenuButtonView,
    });
    /**
     * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
     * future.
     */
    exports.default = exports.MenuButton;
});
