// @uifabric/azure-themes/lib/azure/styles/PrimaryButton.styles.js
import * as StyleConstants from '../Constants';
export var PrimaryButtonStyles = function (theme) {
    var semanticColors = theme.semanticColors;
    var extendedSemanticColors = semanticColors;
    return {
        root: {
            backgroundColor: semanticColors.primaryButtonBackground,
            height: StyleConstants.inputControlHeight,
            padding: '0px 16px',
            color: semanticColors.primaryButtonText,
            border: StyleConstants.borderWidth + " solid " + semanticColors.primaryButtonBorder,
            selectors: {
                '::after': {
                    outlineColor: 'white !important',
                },
            },
        },
        rootDisabled: {
            backgroundColor: semanticColors.primaryButtonBackgroundDisabled,
            color: semanticColors.primaryButtonTextDisabled,
            border: StyleConstants.borderWidth + " solid " + extendedSemanticColors.primaryButtonBorderDisabled + " !important",
        },
        rootFocused: {
            selectors: {
                '::after': {
                    outlineColor: semanticColors.primaryButtonText + " !important",
                },
            },
            backgroundColor: semanticColors.primaryButtonBackground,
            color: semanticColors.primaryButtonText,
            borderColor: extendedSemanticColors.primaryCompoundButtonBorder,
        },
        rootHovered: {
            backgroundColor: semanticColors.primaryButtonBackgroundHovered,
            color: semanticColors.primaryButtonTextHovered,
            borderColor: semanticColors.primaryButtonBackgroundHovered,
        },
        rootPressed: {
            backgroundColor: semanticColors.primaryButtonBackgroundPressed,
            color: semanticColors.primaryButtonTextPressed,
            borderColor: semanticColors.primaryButtonBackgroundPressed,
        },
        rootChecked: {
            backgroundColor: semanticColors.primaryButtonBackgroundPressed,
            color: semanticColors.primaryButtonTextPressed,
            border: 'none',
        },
        rootCheckedHovered: {
            backgroundColor: semanticColors.primaryButtonBackgroundHovered,
            color: semanticColors.primaryButtonTextHovered,
        },
        rootCheckedPressed: {
            backgroundColor: semanticColors.primaryButtonBackgroundPressed,
            color: semanticColors.primaryButtonTextPressed,
        },
    };
};
