// @uifabric/experiments/lib-amd/components/Button/Actionable/Actionable.js
define(["require", "exports", "@uifabric/foundation/lib/next/composed", "./Actionable.styles", "./Actionable.state", "./Actionable.view"], function (require, exports, composed_1, Actionable_styles_1, Actionable_state_1, Actionable_view_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /* eslint-disable deprecation/deprecation */
    /**
     * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
     * future.
     */
    exports.Actionable = composed_1.composed({
        displayName: 'Actionable',
        slots: Actionable_view_1.ActionableSlots,
        state: Actionable_state_1.useActionableState,
        styles: Actionable_styles_1.ActionableStyles,
        tokens: Actionable_styles_1.ActionableTokens,
        view: Actionable_view_1.ActionableView,
    });
    /**
     * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
     * future.
     */
    exports.default = exports.Actionable;
});
