// @uifabric/theme-samples/lib-amd/DarkCustomizations/DarkCustomizations.js
define(["require", "exports", "@fluentui/theme", "./styles/DatePickerStyles", "./styles/PeoplePickerStyles", "@uifabric/variants"], function (require, exports, theme_1, DatePickerStyles_1, PeoplePickerStyles_1, variants_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var DarkDefaultPalette = {
        themeDarker: '#82c7ff',
        themeDark: '#6cb8f6',
        themeDarkAlt: '#3aa0f3',
        themePrimary: '#2899f5',
        themeSecondary: '#0078d4',
        themeTertiary: '#235a85',
        themeLight: '#004c87',
        themeLighter: '#043862',
        themeLighterAlt: '#092c47',
        black: '#ffffff',
        neutralDark: '#faf9f8',
        neutralPrimary: '#f3f2f1',
        neutralPrimaryAlt: '#c8c6c4',
        neutralSecondary: '#a19f9d',
        neutralSecondaryAlt: '#979693',
        neutralTertiary: '#797775',
        neutralTertiaryAlt: '#484644',
        neutralQuaternary: '#3b3a39',
        neutralQuaternaryAlt: '#323130',
        neutralLight: '#292827',
        neutralLighter: '#252423',
        neutralLighterAlt: '#201f1e',
        white: '#1b1a19',
        redDark: '#F1707B',
    };
    exports.PersonaCoinStyles = function (props) {
        return {
            initials: {
                color: props.showUnknownPersonaCoin ? exports.DarkTheme.palette.redDark : exports.DarkTheme.palette.black,
            },
        };
    };
    exports.DarkTheme = theme_1.createTheme({
        palette: DarkDefaultPalette,
        semanticColors: {
            buttonText: DarkDefaultPalette.black,
            buttonTextPressed: DarkDefaultPalette.neutralDark,
            buttonTextHovered: DarkDefaultPalette.neutralPrimary,
            disabledBackground: DarkDefaultPalette.neutralQuaternaryAlt,
            inputBackgroundChecked: DarkDefaultPalette.themePrimary,
            menuBackground: DarkDefaultPalette.neutralLighter,
            menuItemBackgroundHovered: DarkDefaultPalette.neutralQuaternaryAlt,
            menuItemBackgroundPressed: DarkDefaultPalette.neutralQuaternary,
            menuDivider: DarkDefaultPalette.neutralTertiaryAlt,
            menuIcon: DarkDefaultPalette.themeDarkAlt,
            menuHeader: DarkDefaultPalette.black,
            menuItemText: DarkDefaultPalette.neutralPrimary,
            menuItemTextHovered: DarkDefaultPalette.neutralDark,
        },
        isInverted: true,
    });
    var componentStyles = {
        Card: {
            styles: {
                root: {
                    background: exports.DarkTheme.palette.neutralLighter,
                },
            },
        },
        DatePicker: {
            styles: DatePickerStyles_1.DatePickerStyles,
        },
        DetailsList: {
            styles: {
                headerWrapper: {
                    selectors: {
                        '.ms-DetailsHeader': {
                            borderColor: exports.DarkTheme.palette.neutralQuaternary,
                        },
                    },
                },
            },
        },
        ActionButton: {
            styles: {
                root: {
                    backgroundColor: exports.DarkTheme.palette.white,
                },
                rootDisabled: {
                    backgroundColor: exports.DarkTheme.palette.neutralLighter,
                },
                rootHovered: {
                    backgroundColor: exports.DarkTheme.palette.neutralLight,
                },
                rootPressed: {
                    backgroundColor: exports.DarkTheme.palette.neutralQuaternaryAlt,
                },
            },
        },
        DetailsRow: {
            styles: {
                root: {
                    selectors: {
                        ':hover': {
                            background: exports.DarkTheme.palette.neutralLighter,
                        },
                    },
                    borderColor: exports.DarkTheme.palette.neutralQuaternaryAlt,
                },
            },
        },
        Modal: {
            styles: {
                main: {
                    backgroundColor: exports.DarkTheme.palette.neutralLighter,
                },
            },
        },
        Overlay: {
            styles: {
                root: {
                    background: exports.DarkTheme.palette.blackTranslucent40,
                },
            },
        },
        VerticalDivider: {
            styles: {
                divider: {
                    backgroundColor: exports.DarkTheme.palette.neutralQuaternaryAlt,
                },
                wrapper: {
                    Backgroundcolor: exports.DarkTheme.palette.green,
                },
            },
        },
        DocumentCard: {
            styles: {
                root: {
                    border: "1px solid " + exports.DarkTheme.palette.neutralQuaternaryAlt,
                    selectors: {
                        '.ms-DocumentCardPreview': {
                            borderRight: "1px solid " + exports.DarkTheme.palette.neutralQuaternaryAlt,
                        },
                    },
                },
            },
        },
        DocumentCardPreview: {
            styles: {
                root: {
                    borderBottom: "1px solid " + exports.DarkTheme.palette.neutralQuaternaryAlt,
                    borderRight: "1px solid " + exports.DarkTheme.palette.neutralQuaternaryAlt,
                },
            },
        },
        Panel: {
            styles: {
                main: {
                    backgroundColor: exports.DarkTheme.palette.neutralLighter,
                },
                closeButton: {
                    color: exports.DarkTheme.palette.neutralSecondary,
                    selectors: {
                        ':hover': {
                            color: exports.DarkTheme.palette.neutralPrimary,
                        },
                    },
                },
            },
        },
        PeoplePickerItem: {
            styles: PeoplePickerStyles_1.PeoplePickerItemStyles,
        },
        PersonaCoin: {
            styles: exports.PersonaCoinStyles,
        },
        Separator: {
            styles: {
                root: {
                    selectors: {
                        ':before': {
                            backgroundColor: exports.DarkTheme.palette.neutralQuaternaryAlt,
                        },
                        ':after': {
                            backgroundColor: exports.DarkTheme.palette.neutralQuaternaryAlt,
                        },
                    },
                },
            },
        },
        SpinButton: {
            styles: {
                inputTextSelected: {
                    color: exports.DarkTheme.palette.black,
                    background: exports.DarkTheme.palette.themePrimary,
                },
            },
        },
    };
    exports.DarkTheme.components = componentStyles;
    variants_1.addVariants(exports.DarkTheme);
    exports.DarkCustomizations = {
        settings: {
            theme: exports.DarkTheme,
        },
        scopedSettings: componentStyles,
    };
});
