// @uifabric/theme-samples/lib-commonjs/DarkCustomizations/styles/PeoplePickerStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DarkCustomizations_1 = require("../DarkCustomizations");
exports.PeoplePickerItemStyles = function (props) {
    var selected = props.selected;
    return {
        root: [
            {
                background: DarkCustomizations_1.DarkTheme.palette.neutralQuaternaryAlt,
                selectors: {
                    ':hover': {
                        background: DarkCustomizations_1.DarkTheme.palette.neutralQuaternary,
                    },
                },
            },
            selected && {
                backgroundColor: DarkCustomizations_1.DarkTheme.palette.themeSecondary,
                selectors: {
                    ':hover': {
                        background: DarkCustomizations_1.DarkTheme.palette.themeSecondary,
                    },
                },
            },
        ],
        removeButton: [
            {
                background: DarkCustomizations_1.DarkTheme.palette.neutralQuaternaryAlt,
                color: DarkCustomizations_1.DarkTheme.palette.neutralDark,
                selectors: {
                    ':hover': {
                        background: DarkCustomizations_1.DarkTheme.palette.neutralQuaternary,
                        color: DarkCustomizations_1.DarkTheme.palette.black,
                    },
                },
            },
            selected && {
                background: DarkCustomizations_1.DarkTheme.palette.themeSecondary,
                selectors: {
                    ':hover': {
                        background: DarkCustomizations_1.DarkTheme.palette.themeDark,
                    },
                    ':active': {
                        color: DarkCustomizations_1.DarkTheme.palette.black,
                    },
                },
            },
        ],
        subComponentStyles: {
            persona: {
                primaryText: [
                    {
                        color: DarkCustomizations_1.DarkTheme.palette.neutralPrimary,
                    },
                    selected && {
                        color: DarkCustomizations_1.DarkTheme.palette.black,
                        selectors: {
                            ':hover': {
                                color: DarkCustomizations_1.DarkTheme.palette.black,
                            },
                        },
                    },
                ],
            },
        },
    };
};
