// @uifabric/experiments/lib-commonjs/components/FloatingSuggestionsComposite/FloatingSuggestions.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var styling_1 = require("@uifabric/styling");
var GlobalClassNames = {
    root: 'ms-FloatingSuggestions',
    callout: 'ms-FloatingSuggestions-callout',
};
exports.getStyles = function (props) {
    var _a;
    var theme = styling_1.getTheme();
    if (!theme) {
        throw new Error('theme is undefined or null in Editing item getStyles function.');
    }
    // const { semanticColors } = theme;
    var classNames = styling_1.getGlobalClassNames(GlobalClassNames, theme);
    return {
        root: [classNames.root, {}],
        callout: [
            classNames.callout,
            {
                selectors: (_a = {},
                    _a['.ms-FloatingSuggestionsItem-itemButton'] = {
                        padding: '0px',
                        border: 'none',
                    },
                    _a['.ms-FloatingSuggestionsList'] = {
                        minWidth: '260px',
                    },
                    _a),
            },
        ],
    };
};
