// @uifabric/experiments/lib-commonjs/components/Button/Actionable/Actionable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Temporary import file to experiment with next version of foundation.
var composed_1 = require("@uifabric/foundation/lib-commonjs/next/composed");
var Actionable_styles_1 = require("./Actionable.styles");
var Actionable_state_1 = require("./Actionable.state");
var Actionable_view_1 = require("./Actionable.view");
/* eslint-disable deprecation/deprecation */
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
exports.Actionable = composed_1.composed({
    displayName: 'Actionable',
    slots: Actionable_view_1.ActionableSlots,
    state: Actionable_state_1.useActionableState,
    styles: Actionable_styles_1.ActionableStyles,
    tokens: Actionable_styles_1.ActionableTokens,
    view: Actionable_view_1.ActionableView,
});
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
exports.default = exports.Actionable;
