// @uifabric/experiments/lib-amd/components/Button/SplitButton/SplitButton.js
define(["require", "exports", "@uifabric/foundation/lib/next/composed", "./SplitButton.state", "./SplitButton.styles", "./SplitButton.view"], function (require, exports, composed_1, SplitButton_state_1, SplitButton_styles_1, SplitButton_view_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /* eslint-disable deprecation/deprecation */
    /**
     * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
     * future.
     */
    exports.SplitButton = composed_1.composed({
        displayName: 'SplitButton',
        slots: SplitButton_view_1.SplitButtonSlots,
        state: SplitButton_state_1.useSplitButtonState,
        styles: SplitButton_styles_1.SplitButtonStyles,
        tokens: SplitButton_styles_1.SplitButtonTokens,
        view: SplitButton_view_1.SplitButtonView,
    });
    /**
     * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
     * future.
     */
    exports.default = exports.SplitButton;
});
