// @uifabric/experiments/lib-commonjs/components/Button/ButtonVariants/DefaultButton.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Temporary import file to experiment with next version of foundation.
var composed_1 = require("@uifabric/foundation/lib-commonjs/next/composed");
var Button_state_1 = require("../Button.state");
var Button_styles_1 = require("../Button.styles");
var Button_view_1 = require("../Button.view");
/* eslint-disable deprecation/deprecation */
/** @deprecated */
exports.DefaultButtonTokens = function (props, theme) { return [
    Button_styles_1.baseTokens,
    !!props.href && Button_styles_1.hrefTokens,
    !props.disabled && Button_styles_1.enabledTokens,
    props.circular && Button_styles_1.circularTokens,
    props.checked && Button_styles_1.checkedTokens,
    props.disabled && Button_styles_1.disabledTokens,
]; };
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
exports.DefaultButton = composed_1.composed({
    displayName: 'DefaultButton',
    slots: Button_view_1.ButtonSlots,
    state: Button_state_1.useButtonState,
    styles: Button_styles_1.ButtonStyles,
    tokens: exports.DefaultButtonTokens,
    view: Button_view_1.ButtonView,
});
