// @uifabric/experiments/lib-commonjs/components/Button/ButtonVariants/MessageBarButton.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Temporary import file to experiment with next version of foundation.
var composed_1 = require("@uifabric/foundation/lib-commonjs/next/composed");
var Button_state_1 = require("../Button.state");
var Button_styles_1 = require("../Button.styles");
var Button_view_1 = require("../Button.view");
/* eslint-disable deprecation/deprecation */
var baseTokens = function (props, theme) {
    var palette = theme.palette;
    return {
        borderColor: palette.neutralTertiaryAlt,
        borderColorHovered: palette.neutralTertiaryAlt,
        borderColorPressed: palette.neutralTertiaryAlt,
        contentPadding: 0,
        cursor: 'pointer',
        height: 24,
        minHeight: 24,
        minWidth: 84,
        width: 84,
    };
};
var disabledTokens = function (props, theme) {
    var semanticColors = theme.semanticColors;
    return {
        borderColor: semanticColors.buttonBorderDisabled,
        borderColorHovered: semanticColors.buttonBorderDisabled,
        borderColorPressed: semanticColors.buttonBorderDisabled,
        cursor: 'default',
    };
};
/** @deprecated */
exports.MessageBarButtonTokens = function (props, theme) {
    var regularTokens = Button_styles_1.ButtonTokens(props, theme);
    return [regularTokens, baseTokens, props.disabled && disabledTokens];
};
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
exports.MessageBarButton = composed_1.composed({
    displayName: 'MessageBarButton',
    slots: Button_view_1.ButtonSlots,
    state: Button_state_1.useButtonState,
    styles: Button_styles_1.ButtonStyles,
    tokens: exports.MessageBarButtonTokens,
    view: Button_view_1.ButtonView,
});
