// @uifabric/experiments/lib-commonjs/components/Button/MenuButton/MenuButton.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Temporary import file to experiment with next version of foundation.
var composed_1 = require("@uifabric/foundation/lib-commonjs/next/composed");
var MenuButton_state_1 = require("./MenuButton.state");
var MenuButton_styles_1 = require("./MenuButton.styles");
var MenuButton_view_1 = require("./MenuButton.view");
/* eslint-disable deprecation/deprecation */
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
exports.MenuButton = composed_1.composed({
    displayName: 'MenuButton',
    slots: MenuButton_view_1.MenuButtonSlots,
    state: MenuButton_state_1.useMenuButtonState,
    styles: MenuButton_styles_1.MenuButtonStyles,
    tokens: MenuButton_styles_1.MenuButtonTokens,
    view: MenuButton_view_1.MenuButtonView,
});
/**
 * @deprecated This component was experimental and is not longer being developed on, nor will it be supported in the
 * future.
 */
exports.default = exports.MenuButton;
