// @uifabric/experiments/lib/components/FloatingSuggestionsComposite/FloatingPeopleSuggestions/FloatingPeopleSuggestionItems/SuggestionItemDefault.styles.js
import { getGlobalClassNames, getTheme } from '@uifabric/styling';
var GlobalClassNames = {
    root: 'ms-FloatingPeopleSuggestions-SuggestionItemDefault-persona',
    callout: 'ms-FloatingPeopleSuggestions-SuggestionItemDefault-personaContent',
};
export var getStyles = function (props) {
    var _a;
    var theme = getTheme();
    if (!theme) {
        throw new Error('theme is undefined or null in Editing item getStyles function.');
    }
    // const { semanticColors } = theme;
    var classNames = getGlobalClassNames(GlobalClassNames, theme);
    return {
        persona: [
            classNames.root,
            {
                width: '100%',
                selectors: (_a = {},
                    _a['.ms-Persona-detail'] = {
                        minHeight: '40px',
                    },
                    _a),
            },
        ],
        personaContent: [
            classNames.callout,
            {
                display: 'flex',
                width: '100%',
                justifyContent: 'space-between',
                alignItems: 'center',
                padding: '7px 12px',
            },
        ],
    };
};
