// @uifabric/experiments/lib/components/FloatingSuggestionsComposite/FloatingSuggestionsList/FloatingSuggestionsList.styles.js
import { getGlobalClassNames, getTheme } from '@uifabric/styling';
var GlobalClassNames = {
    root: 'ms-FloatingSuggestionsList',
    suggestionsContainer: 'ms-FloatingSuggestionsList-container',
    title: 'ms-FloatingSuggestionsList-title',
    noSuggestions: 'ms-FloatingSuggestionsList-noSuggestions',
};
export var getStyles = function (props) {
    var theme = getTheme();
    if (!theme) {
        throw new Error('theme is undefined or null in Editing item getStyles function.');
    }
    var palette = theme.palette, fonts = theme.fonts;
    var themePrimary = palette.themePrimary, neutralLight = palette.neutralLight, neutralPrimaryAlt = palette.neutralPrimaryAlt;
    var classNames = getGlobalClassNames(GlobalClassNames, theme);
    return {
        root: [
            classNames.root,
            {
                minWidth: '260px',
            },
        ],
        suggestionsContainer: [
            classNames.suggestionsContainer,
            {
                overflowX: 'auto',
                overflowY: 'auto',
                maxHeight: '300px',
                borderBottom: "1px solid " + neutralLight,
            },
        ],
        title: [
            classNames.title,
            {
                padding: '0 12px',
                color: themePrimary,
                fontSize: fonts.small.fontSize,
                lineHeight: '40px',
                borderBottom: "1px solid " + neutralLight,
            },
        ],
        noSuggestions: [
            classNames.noSuggestions,
            {
                textAlign: 'center',
                color: neutralPrimaryAlt,
                fontSize: fonts.small.fontSize,
                lineHeight: '30px',
            },
        ],
    };
};
