// @uifabric/experiments/lib-commonjs/components/Sidebar/SidebarButton.styles.js
"use strict";
/*!
 * Copyright (C) Microsoft Corporation. All rights reserved.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var Styling_1 = require("office-ui-fabric-react/lib-commonjs/Styling");
var Utilities_1 = require("office-ui-fabric-react/lib-commonjs/Utilities");
var Sidebar_styles_1 = require("./Sidebar.styles");
var Sidebar_types_1 = require("./Sidebar.types");
exports.getSidebarButtonStyles = Utilities_1.memoizeFunction(function (theme, sidebarButtonStyles, customStyles) {
    var buttonStyles = {
        root: {
            width: '100%',
            minWidth: Sidebar_types_1.SidebarStylingConstants.sidebarCollapsedWidth,
            padding: '0',
            minHeight: '48px',
            height: 'auto',
            lineHeight: '20px',
            paddingLeft: '4px',
            border: '0',
        },
        rootChecked: {
            borderLeft: "4px solid " + theme.palette.themePrimary,
            paddingLeft: '0',
        },
        rootCheckedHovered: {
            borderLeft: "4px solid " + theme.palette.themePrimary,
            paddingLeft: '0',
        },
        rootCheckedPressed: {
            borderLeft: "4px solid " + theme.palette.themePrimary,
            paddingLeft: '0',
        },
        flexContainer: {
            justifyContent: 'flex-start',
            minHeight: '48px',
            height: 'auto',
        },
        icon: {
            marginLeft: '12px',
            marginRight: '16px',
            fontSize: Sidebar_types_1.SidebarStylingConstants.sidebarIconSize,
            height: Sidebar_types_1.SidebarStylingConstants.sidebarIconSize,
            width: Sidebar_types_1.SidebarStylingConstants.sidebarIconSize,
        },
        labelChecked: {
            fontWeight: 'bold',
            fontFamily: Sidebar_styles_1.sidebarFonts.segoeUiSemibold,
        },
        labelDisabled: {
            fontFamily: Sidebar_styles_1.sidebarFonts.segoeUiSemilight,
        },
        label: {
            verticalAlign: 'middle',
            textAlign: 'left',
            fontSize: '14px',
            whiteSpace: 'normal',
            display: 'inline-block',
            fontFamily: Sidebar_styles_1.sidebarFonts.segoeUiSemilight,
            float: 'left',
            lineHeight: '20px',
            paddingRight: '8px',
            fontWeight: 'normal',
        },
        menuIcon: {
            marginLeft: '16px',
            marginRight: '16px',
            fontSize: Sidebar_types_1.SidebarStylingConstants.sidebarIconSize,
            height: Sidebar_types_1.SidebarStylingConstants.sidebarIconSize,
            width: Sidebar_types_1.SidebarStylingConstants.sidebarIconSize,
        },
    };
    return Styling_1.concatStyleSets(buttonStyles, sidebarButtonStyles, customStyles);
});
exports.getSidebarChildrenStyles = Utilities_1.memoizeFunction(function (theme, customStyles) {
    var sidebarChildrenStyles = {
        flexContainer: {
            paddingLeft: '44px',
        },
        icon: {
            padding: '0',
            marginLeft: '0',
        },
    };
    return Styling_1.concatStyleSets(exports.getSidebarButtonStyles(theme), sidebarChildrenStyles, customStyles);
});
